dev/tasks/main.go

```diff
@@ -0,0 +1,283 @@
+
+package main
+
+//  go run . --remote upstream  --branch upstream/master --version-file version/VERSION  --source ${REPO} -v=2 --yes=true
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	// Support GCP auth
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/klog/v2"
+)
+
+func main() {
+	ctx := context.Background()
+	err := run(ctx)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func run(ctx context.Context) error {
+	log := klog.FromContext(ctx)
+
+	remote := ""
+	flag.StringVar(&remote, "remote", remote, "name of remote")
+	sourceCheckout := ""
+	flag.StringVar(&sourceCheckout, "source", sourceCheckout, "location of source code checkout")
+	branch := ""
+	flag.StringVar(&branch, "branch", branch, "branch for creation of tags")
+	versionFile := ""
+	flag.StringVar(&versionFile, "version-file", versionFile, "path to version marker file")
+	pushOptions := ""
+	flag.StringVar(&pushOptions, "push-options", pushOptions, "when pushing tags to git, include these options")
+	yes := false
+	flag.BoolVar(&yes, "yes", yes, "actually do it")
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	if branch == "" {
+		return fmt.Errorf("must specify --branch")
+	}
+
+	if remote == "" {
+		return fmt.Errorf("must specify --remote")
+	}
+
+	if sourceCheckout == "" {
+		return fmt.Errorf("must specify --source")
+	}
+
+	if versionFile == "" {
+		return fmt.Errorf("must specify --version-file")
+	}
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("error getting home directory: %w", err)
+	}
+	sourceCheckout = strings.ReplaceAll(sourceCheckout, "~", homeDir)
+
+	tmpDir, err := os.MkdirTemp("", "worktree")
+	if err != nil {
+		return fmt.Errorf("error creating temp directory: %w", err)
+	}
+	defer func() {
+		if err := os.RemoveAll(tmpDir); err != nil {
+			klog.Warningf("error removing temp directory: %v", err)
+		}
+	}()
+
+	originalWorktree := gitWorktree{dir: sourceCheckout}
+
+	currentSHA, err := originalWorktree.RevParse(ctx, branch)
+	if err != nil {
+		return err
+	}
+	currentVersionBytes, err := originalWorktree.ReadFile(ctx, currentSHA, versionFile)
+	if err != nil {
+		return err
+	}
+	lastVersionBytes, err := originalWorktree.ReadFile(ctx, currentSHA+"^", versionFile)
+	if err != nil {
+		return err
+	}
+
+	currentVersion := string(currentVersionBytes)
+	currentVersion = strings.TrimSpace(currentVersion)
+	lastVersion := string(lastVersionBytes)
+	lastVersion = strings.TrimSpace(lastVersion)
+
+	if currentVersion == lastVersion {
+		log.Info("no version change; nothing to do", "currentVersion", currentVersion, "lastVersion", lastVersion)
+		return nil
+	}
+
+	// TODO: Check tag does not exist?
+	// originalRefs, err := originalWorktree.ShowRefs(ctx)
+	// if err != nil {
+	// 	return fmt.Errorf("error showing refs: %w", err)
+	// }
+
+	message := "Release " + currentVersion
+	tagName := currentVersion
+	if addVPrefix {
```

Comment:
As the linter points out this doesn't seem to be defined. Do we have anything which builds this?